process/outportBlockConverter.go

```diff
@@ -40,6 +42,349 @@ func NewOutportBlockConverter(
 	}, nil
 }
 
+func (o *outportBlockConverter) HandleShardOutportBlockV2(outportBlock *outport.OutportBlock) (*hyperOutportBlocks.ShardOutportBlockV2, error) {
+	headerType := outportBlock.BlockData.HeaderType
+
+	// check if the header type is supported by this function.
+	if headerType != string(core.ShardHeaderV1) && headerType != string(core.ShardHeaderV2) {
+		return nil, fmt.Errorf("cannot convert to shard outport block. header type: %s not supported", headerType)
+	}
+
+	shardOutportBlock := &hyperOutportBlocks.ShardOutportBlockV2{}
+	shardOutportBlock.ShardID = outportBlock.ShardID
+	err := o.handleBlockData(outportBlock.BlockData, shardOutportBlock)
+	if err != nil {
+		return nil, fmt.Errorf("failed to migrate block data: %w", err)
+	}
+	err = o.handleTransactionPool(outportBlock.TransactionPool, shardOutportBlock)
+	if err != nil {
+		return nil, fmt.Errorf("failed to migrate transacion pool: %w", err)
+	}
+	handleHeaderGasConsumption(outportBlock.HeaderGasConsumption, shardOutportBlock)
+	handleAlteredAccounts(outportBlock.AlteredAccounts, shardOutportBlock)
+
+	shardOutportBlock.NotarizedHeadersHashes = outportBlock.NotarizedHeadersHashes
+	shardOutportBlock.NumberOfShards = outportBlock.NumberOfShards
+	shardOutportBlock.SignersIndexes = outportBlock.SignersIndexes
+	shardOutportBlock.HighestFinalBlockNonce = outportBlock.HighestFinalBlockNonce
+	shardOutportBlock.HighestFinalBlockHash = outportBlock.HighestFinalBlockHash
+
+	return shardOutportBlock, nil
+}
+
+func (o *outportBlockConverter) handleBlockData(blockData *outport.BlockData, shardOutportBlock *hyperOutportBlocks.ShardOutportBlockV2) error {
+	if blockData == nil {
```

Comment:
is there a case where blockData can be nil? if yes, shouldn't we return an error if that's happending, instead of returning nil?

Reply:
I don't remember why I put that check. But I'll switch to error.